### ISSUE TITLE
run_built_binary fails on linux

### ISSUE THREAD

**Issue by BreD1810** [herd/herdtools7#150]
Line 29 makes a call to `stat` using `-f`. On Linux, this displays file system status, so the following error occurs:
```bash
make test                                                                                                                   

...

internal/run_built_binary herd_catalogue_regression_test \
	-herd-path _build/install/default/bin/herd7 \
	-libdir-path herd/libdir \
	-kinds-dir catalogue/aarch64/kinds \
	-shelf-path catalogue/aarch64/shelf.py \
	test
stat: cannot read file system information for '%m %N': No such file or directory
internal/run_built_binary: line 45: Total:: command not found
make: *** [Makefile:110: test] Error 127
```

I believe the option is supposed to be for formatting, so perhaps `--format` is preferred?

The man page is available [here](https://linux.die.net/man/1/stat).

Once I apply this fix, the script still fails, this time with:
```bash
internal/run_built_binary herd_catalogue_regression_test \
	-herd-path _build/install/default/bin/herd7 \
	-libdir-path herd/libdir \
	-kinds-dir catalogue/aarch64/kinds \
	-shelf-path catalogue/aarch64/shelf.py \
	test
internal/run_built_binary: line 45: '/home/brad/gdp/repo/_build/default/internal/herd_catalogue_regression_test.exe': No such file or directory
make: *** [Makefile:110: test] Error 127
```

This file does exist, but apostrophes are not stripped from around the filename, so it cannot find the file.
I can make the test pass by changing line 45 to:
```bash
command="$(most_recent ${paths} | xargs) $@"
$command
```
I'm not the best with Bash, so perhaps there is a better way.

**Comment by BreD1810** [comment]
It seems that this script also fails with the shell "dash", which I think is `/bin/sh` on Debian systems (it certainly is in our CI docker image).

Running the script gets the following error:
```bash
internal/run_built_binary: 17: internal/run_built_binary: Syntax error: "(" unexpected
make: *** [Makefile:110: test] Error 2
```

The `function` keyword is a ["bashism"](https://wiki.ubuntu.com/DashAsBinSh#function), hence breaking non-bash compatible shells like dash. Simply removing them fixes the script whilst maintaining bash support:
```bash
absolute_path() {
  readonly prev_pwd="${PWD}"
  cd "${1}"
  pwd
  cd "${prev_pwd}"
}

most_recent() {
  # List (mtime, path) pairs.
  # Sort them descending by mtime.
  # Take the first of the list.
  # Extract the path back out.
  stat -s '%m %N' $@ \
    | sort --reverse \
    | head -n1 \
    | cut -d' ' -f2
}
```

**Comment by eth-arm** [comment]
The original issue was that `stat`'s options vary on different platforms (https://shellhaters.org is a reference of what binaries and options _are_ portable).

Good catch on the bashisms, however.

PR #153 should fix this and another issue.

**Comment by BreD1810** [comment]
PR #153 works for me, thanks!

**Comment by BreD1810** [comment]
Resolved by #153
